### ISSUE TITLE
Varargs is broken in 1.5.1

### ISSUE THREAD

**Issue by brettwooldridge** [briandilley/jsonrpc4j#198]
@kostapc Dude, #172 broke *all* vararg endpoints *except* ``Object...``.

Please add the following methods to your ``JsonRpcServerAnnotateMethodVarArgsTest`` service ``ServiceInterfaceWithCustomMethodNameWithVarArgsAnnotation`` interface and test.

```java
      @JsonRpcMethod("testMethodVararg2")
      String testMethodVararg2(@WebParam(name = "strings") String... strings);

      @JsonRpcMethod("testMethodVararg3")
      String testMethodVararg3(@WebParam(name = "ints") int... ints);
```

@gaborbernat

**Comment by mslipper** [comment]
Well that's not good. I'll fix tonight.

**Comment by brettwooldridge** [comment]
@mslipper Thanks.  In my brief debugging foray, it looked like because named parameters (as opposed to positional) have a type of JSON_OBJECT, somehow the array ends up getting converted as a ``List`` (as in ``java.util.List``) rather than a native array.  The result is that the endpoint method is invoked with said List while the signature is clearly, for example, ``String[]`` -- which results in a runtime error.

Adding the above tests should quickly surface the source of the error.

**Comment by mslipper** [comment]
Got it. It looks like varargs are in general broken when used with classes other than `Object...`. I'll fix this issue first then move onto a larger fix. I'd appreciate it if you could help me review the PR when I'm finished.

**Comment by brettwooldridge** [comment]
Sure.  It was definitely working in 1.5.0 ... we have a lot of varargs methods.

**Comment by mslipper** [comment]
Released in 1.5.2.
